pkg/apis/planetscale/v2/vitessbackupschedule_types.go

```diff
@@ -0,0 +1,206 @@
+	// runs until Suspend is set to false again. This is useful if you want to pause backup without
+	// having to remove the entire VitessBackupSchedule object from the cluster.
+	// +optional
+	Suspend *bool `json:"suspend,omitempty"`
+
+	// StartingDeadlineSeconds allows for the VitessBackupSchedule controller to start jobs that late
+	// by the given amount of seconds.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	StartingDeadlineSeconds *int64 `json:"startingDeadlineSeconds,omitempty"`
+
+	// ConcurrencyPolicy specifies ho to treat concurrent executions of a Job.
+	// Valid values are:
+	// - "Allow" (default): allows CronJobs to run concurrently;
+	// - "Forbid": forbids concurrent runs, skipping next run if previous run hasn't finished yet;
+	// - "Replace": cancels currently running job and replaces it with a new one.
+	// +optional
+	// +kubebuilder:example="Forbid"
+	ConcurrencyPolicy ConcurrencyPolicy `json:"concurrencyPolicy,omitempty"`
+
+	// AllowedMissedRun defines how many missed run of the schedule will be allowed before giving up on finding the last job.
+	// If the operator's clocked is skewed and we end-up missing a certain number of jobs, finding the last
+	// job might be very consuming, depending on the frequency of the schedule and the duration during which
+	// the operator's clock was misbehaving, also depending on how laggy the clock is, we can end-up with thousands
+	// or missed runs. For this reason, AllowedMissedRun, which is set to 100 by default, will allow us to give up finding
+	// the last job and simply wait for the next job on the schedule.
+	// Unless you are experiencing issue with missed runs due to a misconfiguration of the clock, we recommend leaving
+	// this field to its default value.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	AllowedMissedRun *int `json:"allowedMissedRun,omitempty"`
+
+	// JobTimeoutMinute defines after how many minutes a job that has not yet finished should be stopped and removed.
+	// Default value is 10 minutes.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default=10
+	JobTimeoutMinute int32 `json:"jobTimeoutMinute,omitempty"`
+}
+
+// VitessBackupScheduleStrategy defines how we are going to take a backup.
+// The VitessBackupSchedule controller uses this data to build the vtctldclient
+// command line that will be executed in the Job's pod.
+type VitessBackupScheduleStrategy struct {
+	// Name of the backup strategy.
+	Name BackupStrategyName `json:"name"`
+
+	// KeyspaceShard defines the keyspace and shard on which we want to take a backup.
+	// It has to be formatted as follows: <keyspace>/<shard>
+	// This field is mandatory if we have picked the strategy BackupShard.
+	// This field will be ignored if we have picked the strategy BackupTablet.
+	// +optional
+	// +kubebuilder:example="commerce/-"
+	KeyspaceShard string `json:"keyspaceShard,omitempty"`
```

Comment:
If a shard only has 1 replica, then the `BackupShard` strategy will mean all `@replica` queries will fail. may be worth having an option like `MinHealthyReplicas` for that strategy.

Reply:
The `@replica` will fail because the tablet will not be able to respond query will being backed up? In that case, yeah it would make sense to add such parameter for the user to configure, with a default to 2 for instance.